openlibrary/macros/ReadingLogButton.html

```diff
@@ -1,13 +1,26 @@
-$def with (work, read_status)
+$def with (work, read_status, url=None, searchPage = None)
 
-<form method="POST" action="$(work.key)/bookshelves.json?debug=true" class="reading-log-lite">
-  <select name="bookshelf_id">
-    $ shelf_names = {1: _('Want to Read'), 2: _('Currently Reading'), 3: _('Already Read') }
-    $for shelf_id in [1, 2, 3]:
-      $ is_selected = read_status == shelf_id
-      <option value="$shelf_id" $cond(is_selected, 'selected')>
-        $cond(is_selected, '✓') $shelf_names[shelf_id]
-      </option>
-    <option value="-1">$_('Remove')</option>
-  </select>
-</form>
+$if ctx.user:
+  <form method="POST" action="$(work.key)/bookshelves.json?debug=true" class="reading-log-lite $(searchPage)">
+    <select name="bookshelf_id">
+      $ shelf_names = {1: _('Want to Read'), 2: _('Currently Reading'), 3: _('Already Read') }
+      $for shelf_id in [1, 2, 3]:
+        $ is_selected = read_status == shelf_id
+        <option value="$shelf_id" $cond(is_selected, 'selected')>
+          $cond(is_selected, '✓') $shelf_names[shelf_id]
+        </option>
+      <option value="-1">$_('Remove')</option>
+    </select>
+  </form>
+
+$else:
+  <div class="dropit">
+    <div class="dropper on">
+        <div class="log-work">
+            <button class="unactivated" type="submit">$_('Want to Read')</button>
+            <a href="/account/login?redir=$(url)" class="dropclick-prevent dropclick-unactivated">
+              <div class="arrow arrow-unactivated"></div>
+            </a>
+        </div>
+    </div>
+  </div>
```

Comment:
```suggestion
  </div>

```
Add newline to the end of the file.